.gitlab-ci.example.yml

```diff
@@ -10,21 +10,38 @@
 # * Set in the schedule required variables
 #
 #     PROJECT_PATH = group/repository
-#     PACKAGE_MANAGER_SET = bundler,composer,npm_and_yarn
+#     PACKAGE_MANAGER = bundler,composer,npm_and_yarn
 #
 # https://github.com/dependabot/dependabot-script
 # https://docs.gitlab.com/ee/ci/yaml/
 
+stages:
+  - build
+  - dependabot
+
 build-image:
   stage: build
+  image: docker
+  services:
+    - name: docker:dind
+      command: ["--experimental"]
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "schedule"'
+      when: never
+    - if: '$CI_COMMIT_BRANCH == "main"'
+    - when: never
+  before_script:
+    - docker login -u $CI_REGISTRY_USER -p $CI_REGISTRY_PASSWORD $CI_REGISTRY
```

Comment:
Is there a reason we have to use a custom registry? I'd rather keep the example simple/straightforward, and then advanced users can further customize it.

Using `dependabot/dependabot-script` should be fine AFAIK (although I have no way to test)...

And if we aren't currently publishing `dependabot/dependabot-script` (or something equivalent) then we should... I'd rather solve that in a separate issue and not muddle the example even if it's broken until we get to that... cleaning up some of this build infra / documentation confusion is on my plate for this quarter, so I hopefully can get to it.

Reply:
The principal reason it's the initial do like this. I'm happy to know there is a image on docker hub. I will take that in count and change it

Reply:
I don't know if there is, but if not, we should fix it... but that'll probably be sequenced behind some other related cleanups that need to happen first. So could be a few months depending on how fast I work through things.